sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ColumnDefinition.scala

```diff
@@ -161,43 +162,47 @@ object ColumnDefinition {
         cmd.columns.foreach { col =>
           col.defaultValue.foreach { default =>
             checkDefaultColumnConflicts(col)
+            if (!default.resolved) {
+              throw QueryCompilationErrors.defaultValuesUnresolvedExprError(
+                statement, col.name, default.originalSQL, null)
+            }
             validateDefaultValueExpr(default, statement, col.name, col.dataType)
             if (!default.deterministic) {
               throw QueryCompilationErrors.defaultValueNonDeterministicError(
-                "CREATE TABLE", col.name, default.originalSQL)
+                statement, col.name, default.originalSQL)
             }
           }
         }
 
       case cmd: AddColumns if cmd.columnsToAdd.exists(_.default.isDefined) =>
-        // Wrap analysis errors for default values in a more user-friendly message.
+        val statement = "ALTER TABLE ADD COLUMNS"
```

Comment:
This matches messages like:  https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L298